src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -272,7 +272,15 @@ private void SetupSingleCheck(CheckFactoryContext checkFactoryContext, string pr
                 // Create the wrapper and register to central context
                 wrapper.StartNewProject(projectFullPath, configurations, userEditorConfigs);
                 var wrappedContext = new CheckRegistrationContext(wrapper, _buildCheckCentralContext);
-                check.RegisterActions(wrappedContext);
+                try
+                {
+                    check.RegisterActions(wrappedContext);
+                }
+                catch (Exception e)
+                {
+                    throw new BuildCheckConfigurationException(
```

Comment:
Prefer the `BuildCheckConfigurationException` that takes the innerExcpetion argument - not to loose the original stack of the problem

Reply:
I did not add it before because we already lose the exception, since we log it as a warning. But adding just in case for the future.